grudge/execution.py

```diff
@@ -530,14 +530,33 @@ def prg(nmatrices):
             if in_grp.nelements == 0:
                 continue
 
+            # Cache operator
+            cache_key = ("diff_batch", in_grp, out_grp, repr_op, field.dtype)
```

Comment:
You need to consider all operators in the cache key, not just the first one.

Reply:
Yeah, that seems to do it.